server/server.go

```diff
@@ -128,6 +138,42 @@ func New(cfg Config) (*Server, error) {
 		grpcListener = netutil.LimitListener(grpcListener, cfg.GRPCConnLimit)
 	}
 
+	// If user doesn't supply a logging implementation, by default instantiate
+	// logrus.
+	log := cfg.Log
+	if log == nil {
+		log = logging.NewLogrus(cfg.LogLevel)
+	}
+
+	// Setup TLS
+	var cert tls.Certificate
+	if cfg.HTTPCertPath != "" && cfg.HTTPKeyPath != "" {
+		cert, err = tls.LoadX509KeyPair(cfg.HTTPCertPath, cfg.HTTPKeyPath)
+		if err != nil {
+			log.Warnf("error loading cert %s or key %s, tls disabled", cfg.HTTPCertPath, cfg.HTTPKeyPath)
+		}
+	}
+
+	var caCertPool *x509.CertPool
+	if cfg.HTTPCAPath != "" {
+		caCert, err := ioutil.ReadFile(cfg.HTTPCAPath)
+		if err != nil {
+			log.Warnf("error loading ca cert %s, tls disabled", cfg.HTTPCAPath)
+		} else {
+			caCertPool = x509.NewCertPool()
+			caCertPool.AppendCertsFromPEM(caCert)
+		}
+	}
+
+	var tlsConfig *tls.Config
+	if len(cert.Certificate) > 0 && caCertPool != nil {
+		tlsConfig = &tls.Config{
+			Certificates: []tls.Certificate{cert},
+			ClientAuth:   tls.RequireAndVerifyClientCert,
+			ClientCAs:    caCertPool,
+		}
+	}
```

Comment:
Nit: I would put this all in a separate function which gives you a TLS config - might make it easier to test.